### PR TITLE
chore: renaming gather files

### PR DESCRIPTION
**Pull request by rluders**
This PR renames the gather files  to match gather method and prefix it with gather.

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

N/A

## Documentation

- `docs/gathered-data.md`

## Unit Tests

It also renames the test files to match the gather file.

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-10109
https://issues.redhat.com/browse/CCXDEV-10104
https://issues.redhat.com/browse/CCXDEV-10103
https://issues.redhat.com/browse/CCXDEV-10129

### CODE REVIEW

**Comment by ncaak** [comment]
This could backtrack to these Jira issues, as it partially solves them:
* https://issues.redhat.com/browse/CCXDEV-10103
* https://issues.redhat.com/browse/CCXDEV-10104
* https://issues.redhat.com/browse/CCXDEV-10129

Also, could we take advantage on this PR and sort `pkg/gatherers/clusterconfig/clusterconfig_gatherer.go` _gatheringFunctions_ function items? Maybe they're sorted in a way I cannot comprehend. But, if they're not, I could see some improvement if the items would be sorted alphabetically, instead chaotically.

**Comment by rluders** [comment]
@ncaak sorting the _gathetingFunctions_ is actually a good idea. I'll update the PR.

**Comment by rluders** [comment]
I'm changing it for the draft, 'cause I'll include the unit tests for:

- ClusterOperatos (already exists, I'm just double checking it)
- ClusterAuthentication
- TSBDStatus
- FeatureGates

**Review by ncaak — APPROVED** [review]
/lgtm

**Review by ncaak — APPROVED** [review]
/lgtm

File: pkg/gatherers/clusterconfig/clusterconfig_gatherer.go [hunk 1]
```diff
@@ -27,66 +27,64 @@ type Gatherer struct {
...
+	"jaegers":                           (*Gatherer).GatherJaegerCR,
+	"kube_controller_manager_logs":      (*Gatherer).GatherKubeControllerManagerLogs,
+	"machine_autoscalers":               (*Gatherer).GatherMachineAutoscalers,
 	"machine_configs":                   (*Gatherer).GatherMachineConfigs,
 	"machine_healthchecks":              (*Gatherer).GatherMachineHealthCheck,
+	"machine_sets":                      (*Gatherer).GatherMachineSet,
+	"machines":                          (*Gatherer).GatherMachine,
+	"metrics":                           (*Gatherer).GatherMostRecentMetrics,
 	"monitoring_persistent_volumes":     (*Gatherer).GatherMonitoringPVs,
-	"install_plans":                     (*Gatherer).GatherInstallPlans,
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/clusterconfig_gatherer.go:56]: I think this one is lost.

File: pkg/gatherers/clusterconfig/clusterconfig_gatherer.go [hunk 2]
```diff
@@ -27,66 +27,64 @@ type Gatherer struct {
...
-	"service_accounts":                  (*Gatherer).GatherServiceAccounts,
-	"machine_config_pools":              (*Gatherer).GatherMachineConfigPool,
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/clusterconfig_gatherer.go:58]: And same for this one.

**Review by ncaak — APPROVED** [review]
/lgtm

**Pull request merged**